modules/hyprland/hm.nix

```diff
@@ -24,6 +24,11 @@ in {
   options.stylix.targets.hyprland.enable =
     config.lib.stylix.mkEnableTarget "Hyprland" true;
 
-  config.wayland.windowManager.hyprland.settings =
-    lib.mkIf (config.stylix.enable && config.stylix.targets.hyprland.enable) settings;
+  config =
+    lib.mkIf
+    (config.stylix.enable && config.stylix.targets.hyprland.enable)
+    {
+      wayland.windowManager.hyprland.settings = settings;
+      services.hyprpaper.enable = true;
```

Comment:
Enabling the [`services.hyprpaper` Home Manager module](https://github.com/nix-community/home-manager/blob/release-24.05/modules/services/hyprpaper.nix) without further customization is a no-op.

To avoid confusion, we should assert the `hyprland` wallpaper being set based on `services.hyprpaper.enable` and `stylix.targets.hyprpaper.enable`:

| `services.hyprpaper.enable` | `stylix.targets.hyprpaper.enable` | Is `hyprland` wallpaper set |
|---|---|---|
| `false` | `false` | `false` |
| `false` | `true` | `false` |
| `true` | `false` | `false` |
| `true` | `true` | `true` |

Currently Stylix does not support disabling wallpapers being set, when supported. Until this is supported, `stylix.targets.hyprland.enable && services.hyprpaper.enable && stylix.targets.hyprpaper.enable` should be asserted.

@danth, what do you think?

Reply:
Alternatively, the Hyprland module could set `stylix.targets.hyprpaper.enable = true` so that it creates a conflict unless explicitly overridden.

Reply:
Resolved via https://github.com/danth/stylix/pull/377/commits/896a895c4fe58897aee2391bd40a2594461b7297.

Reply:
This change installed and enabled the hyprpaper service on my machine which I find very surprising. I can live with it but I was happily using wpaperd.

Not sure if this was intentional but I wouldn't expect stylix to install and configure new services.

Reply:
> This change installed and enabled the hyprpaper service on my machine which I find surprising. I can live with it but I was happily using wpaperd.
> 
> Not sure if this was intentional but I wouldn't expect stylix to install and configure new services.

This is a known limitation:

> Currently, Stylix does not support disabling wallpapers being set, when supported.

I opened a tracking issue for this: https://github.com/danth/stylix/issues/442.